### PR TITLE
Fixed removal of equivalent clauses in AST.

### PR DESCRIPTION
**Pull request by azreika**
Fixes issue #1745. In some cases, when two equivalent clauses in the same relation were being removed, a clause might be used after being freed due to the by-value deletion of clauses in programs. This PR fixes that (and adds a test).

### CODE REVIEW

File: tests/evaluation/empty_relations2/empty.dl [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Souffle - A Datalog Compiler
```
Comment (mmcgr) [inline tests/evaluation/empty_relations2/empty.dl:1]: Filename needs to match directory name

**Review by mmcgr — DISMISSED** [review]
Approving, but look into making this a unit test rather than integration.

**Comment by azreika** [comment]
> Approving, but look into making this a unit test rather than integration.

Good point, I'll do this.

**Review by b-scholz — DISMISSED** [review]

File: tests/evaluation/empty_relations2/empty_relations2.err [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Warning: No rules/facts defined for relation empty in file empty.dl at line 14
```
Comment (mmcgr) [inline tests/evaluation/empty_relations2/empty_relations2.err:1]:
```suggestion
Warning: No rules/facts defined for relation empty in file empty_relations2.dl at line 14
```

Comment (azreika) [reply tests/evaluation/empty_relations2/empty_relations2.err:1]: Hahahahahha this typo has plagued my existence, cheers for the catch

**Review by mmcgr — APPROVED** [review]

**Review by mmcgr — APPROVED** [review]

**Pull request merged**